src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2806,6 +2808,20 @@ public override void OnPush()
 
                 if (Todo < _stage._parallelism && !HasBeenPulled(_stage.In))
                     TryPull(_stage.In);
+                return;
+
+                async Task RunTask(Task<TOut> tt)
```

Comment:
Use an `async` local function with an `await` on the end-user `Task<TOut>` instead of using `ContinueWith` - this will result in better error handling and fewer weird task context issues.

Replicates the same work we did on https://github.com/akkadotnet/akka.net/pull/7521